src/main/java/seedu/duke/models/schema/Student.java

```diff
@@ -203,4 +214,136 @@ public ModuleList getModulesPlanned() {
     public void printSchedule(){
         this.schedule.printMainModuleList();
     }
+
+    public void setCurrentSemesterModules() {
+        try {
+            int[] yearAndSem = Parser.parseStudentYear(year);
+            System.out.println(Arrays.toString(yearAndSem));
+            int currSem = ((yearAndSem[0] - 1) * 2) + yearAndSem[1];
+            int[] modulesPerSem = schedule.getModulesPerSem();
+            System.out.println(Arrays.toString(modulesPerSem));
+            ModuleList modulesPlanned = schedule.getModulesPlanned();
+            // schedule.printMainModuleList();
+            // System.out.println("-------------------------------");
+            // printSchedule();
+            int numberOfModulesInCurrSem = modulesPerSem[currSem - 1];
+            int numberOfModulesCleared = 0;
+            for (int i = 0; i < currSem - 1; i++) {
+                numberOfModulesCleared += modulesPerSem[i];
+            }
+            int startIndex = numberOfModulesCleared;
+            int endIndex = startIndex + numberOfModulesInCurrSem;
+            ArrayList<Module> modulesInSchedule = modulesPlanned.getMainModuleList();
+            currentSemesterModules = new ModuleList();
+            for (int i = startIndex; i < endIndex; i++) {
+                currentSemesterModules.addModule(modulesInSchedule.get(i));
+                //          System.out.println(currentSemesterModules.getModule(i));
+                //          System.out.println(modulesInSchedule.get(i).getModuleCode());
+            }
+            ArrayList<Module> toprint = currentSemesterModules.getMainModuleList();
+            for (int i = 0; i < toprint.size(); i++) {
+                System.out.println(toprint.get(i).getModuleCode());
+            }
```

Comment:
Structuring a code logically can make it easier for you and other developers to understand your code, and increase its readability! Empty lines can do a lot....
![image](https://github.com/AY2324S1-CS2113-T17-4/tp/assets/110933149/a1a76d38-2551-44f0-a81c-a0ef852b6700)